x-pack/plugins/entity_manager/public/lib/entity_client.ts

```diff
@@ -95,7 +95,7 @@ export class EntityClient {
     const identityFieldsValue = this.getIdentityFieldsValue(entityInstance);
 
     const nodes: KueryNode[] = Object.entries(identityFieldsValue).map(([identityField, value]) => {
-      return nodeTypes.function.buildNode('is', identityField, value);
+      return nodeTypes.function.buildNode('is', identityField, `"${value}"`);
```

Comment:
To reproduce the error, just replace ``"${value}"`` with `value` and run the entity_client unit tests